Nodejs/Product/Nodejs/Options/NodejsIntellisenseOptionsPage.cs

```diff
@@ -148,6 +169,41 @@ public NodejsIntellisenseOptionsPage()
             SaveString(CompletionCommittedBySetting, CompletionCommittedBy);
             SaveBool(SaveToDiskSetting, SaveToDisk);
 
+        }
+
+        private string _toolsVersion;
+        private string GetTypeScriptToolsVersion() {
+            if (_toolsVersion == null) {
+                _toolsVersion = string.Empty;
+                try {
+                    // Start with the Nodejs tools path to retrieve the IDE path
+                    var idePath = Assembly.GetExecutingAssembly().Location;
+                    while (!string.IsNullOrEmpty(idePath) && !StringComparer.OrdinalIgnoreCase.Equals(Path.GetFileName(idePath), "IDE")) {
+                        var parentInfo = Directory.GetParent(idePath);
+                        idePath = parentInfo != null ? parentInfo.FullName : string.Empty;
+                    }
+
+                    if (string.IsNullOrEmpty(idePath)) {
+                        return _toolsVersion;
+                    }
+
+                    var typeScriptServicesPath = Path.Combine(idePath, @"CommonExtensions\Microsoft\TypeScript\typescriptServices.js");
+                    if (!File.Exists(typeScriptServicesPath)) {
+                        return _toolsVersion;
+                    }
+
+                    var regex = new Regex(@"toolsVersion = ""(?<version>\d.\d?)"";");
+                    var fileText = File.ReadAllText(typeScriptServicesPath);
+                    var match = regex.Match(fileText);
+
+                    var version = match.Groups["version"].Value;
+                    if (!string.IsNullOrWhiteSpace(version)) {
+                        _toolsVersion = version;
+                    }
+                } catch (Exception) { }

```

Comment:
What exception are you expecting here? Debug.WriteLine might be better here rather than completely swallowing the exception.

Also don't swallow any critical exceptions like OutOfMemory, etc. You can use ```ExceptionExtensions.IsCriticalException``` here